File: sbe/binance/spot_3_2/generated-php/spot_sbe/LotSizeFilter.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class LotSizeFilter
{
    public const TEMPLATE_ID = 4;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 2;
    public const BLOCK_LENGTH = 25;

    public int|float|array|null $filterType = null;
    public int|float|array|null $qtyExponent = null;
    public int|float|array|null $minQty = null;
    public int|float|array|null $maxQty = null;
    public int|float|array|null $stepSize = null;

    public function encode(): string
    {
        $buffer = '';

        if ($this->qtyExponent !== null) {
            $buffer .= pack('c', $this->qtyExponent);
        }
        if ($this->minQty !== null) {
            $buffer .= pack('q', $this->minQty);
        }
        if ($this->maxQty !== null) {
            $buffer .= pack('q', $this->maxQty);
        }
        if ($this->stepSize !== null) {
            $buffer .= pack('q', $this->stepSize);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->qtyExponent = unpack('c', substr($data, $offset, 1))[1];
        $offset += 1;
        $this->minQty = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->maxQty = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->stepSize = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
    }
}

File: sbe/binance/spot_3_1/generated-python/spot_sbe/LotSizeFilter.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class LotSizeFilter:
    """SBE message: LotSizeFilter."""

    TEMPLATE_ID = 4
    SCHEMA_ID = 3
    SCHEMA_VERSION = 1
    BLOCK_LENGTH = 25

    def __init__(self):
        self.filter_type = None
        self.qty_exponent = None
        self.min_qty = None
        self.max_qty = None
        self.step_size = None

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.qty_exponent is not None:
            buffer.write(struct.pack('<b', self.qty_exponent))
        if self.min_qty is not None:
            buffer.write(struct.pack('<q', self.min_qty))
        if self.max_qty is not None:
            buffer.write(struct.pack('<q', self.max_qty))
        if self.step_size is not None:
            buffer.write(struct.pack('<q', self.step_size))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        pos = 0

        self.qty_exponent = struct.unpack_from('<b', data, pos)[0]
        pos += 1
        self.min_qty = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.max_qty = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.step_size = struct.unpack_from('<q', data, pos)[0]
        pos += 8

        # Skip to end of block for forward compatibility
        pos = 25

